rust/gui-client/src-tauri/src/ipc.rs

```diff
@@ -181,24 +128,33 @@ impl<E: serde::Serialize> tokio_util::codec::Encoder<&E> for Encoder<E> {
     }
 }
 
-/// Connect to the IPC service
-///
-/// Public because the GUI Client will need it
-pub async fn connect_to_service(id: ServiceId) -> Result<(ClientRead, ClientWrite)> {
+/// Connect to an IPC socket
+pub async fn connect<R, W>(id: SocketId) -> Result<(ClientRead<R>, ClientWrite<W>)>
+where
+    R: DeserializeOwned,
+    W: Serialize,
+{
+    tracing::debug!(
+        ?id,
+        client_pid = std::process::id(),
+        "Connecting to IPC socket"
+    );
+
     // This is how ChatGPT recommended, and I couldn't think of any more clever
     // way before I asked it.
     let mut last_err = None;
 
     for _ in 0..10 {
-        match platform::connect_to_service(id).await {
+        match platform::connect_to_socket(id).await {
             Ok(stream) => {
                 let (rx, tx) = tokio::io::split(stream);
                 let rx = FramedRead::new(rx, Decoder::default());
                 let tx = FramedWrite::new(tx, Encoder::default());
                 return Ok((rx, tx));
             }
+            Err(error) if error.root_cause().is::<NotFound>() => return Err(error), // If the socket isn't there, fail right away.
```

Comment:
It seems that this makes the tests racy :(

I'll have to think of something else.